tests/doubles/fake_console_io.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <algorithm>
+
+#include "greentea-client/test_io.h"
+#include "fake_console_io.h"
+
+static std::string _stdout;
+static std::string _stdin;
+
+Console::Console()
+{
+}
+
+std::string Console::get_stdout()
+{
+    return _stdout;
+}
+
+void Console::set_stdin(const std::string& str)
+{
+    _stdin.append(str);
+    std::reverse(_stdin.begin(), _stdin.end());
+}
+
+Console::~Console()
+{
+    _stdout = {};
+    _stdin = {};
+}
+
+int greentea_getc()
+{
+    int out = EOF;
+    if(!_stdin.empty())
```

Comment:
Do we not put a space after `if`? Is astyle not catching this?

Reply:
I'll take a look. I also thought it was slightly odd that astyle didn't flag anything. The astyle check is a copy-paste from the mbed-os travis job, I'm assuming it worked there.

Reply:
I've fixed the astyle check (it's now failing as expected). Now to fix the formatting!